cdap-kubernetes/src/main/java/io/cdap/cdap/k8s/runtime/KubeTwillController.java

```diff
@@ -145,6 +146,31 @@ public ResourceReport getResourceReport() {
     return doRestartInstances(runnable, instanceIds);
   }
 
+  @Override
+  public Future<String> restartInstance(String runnable, int instanceId, String uid) {
+    CompletableFuture<String> resultFuture = new CompletableFuture<>();
+    CoreV1Api api = new CoreV1Api(apiClient);
+
+    // Only support for stateful set for now
+    if (!V1StatefulSet.class.equals(resourceType)) {
+      resultFuture.completeExceptionally(
+        new UnsupportedOperationException("Instance restart by instance id is only supported for statefulsets"));
+      return resultFuture;
+    }
+
+    // pod uid is passed as "runnable" parameter to the method.
```

Comment:
This comment is no longer valid now right?

Reply:
removed